src/sleep.rs

```diff
@@ -0,0 +1,110 @@
+//! Driver for light and deep sleep.
+//!
+//! Currently implemented Timer and UART wakeup sources.
+
+use esp_idf_sys::*;
+use std::time::Duration;
+
+#[derive(Debug)]
+pub enum WakeupSource {
+    Timer { dur: Duration },
+    Uart { uart_num: i32, threshold: i32 },
+    // TODO: add more sources
+}
+
+impl WakeupSource {
+    pub fn apply(&self) -> Result<(), EspError> {
+        match *self {
+            WakeupSource::Timer { dur } => {
+                esp!(unsafe { esp_sleep_enable_timer_wakeup(dur.as_micros() as u64) })?;
+            }
+            WakeupSource::Uart {
+                uart_num,
+                threshold,
```

Comment:
Shuldn't this be `u16`?

Reply:
according to cargo build it expects a i32

Reply:
Sure, but that's just an oddity of the C API. Looking at the documentation, it should be between 3 and 0x3ff, right? Which is `u16`.